src/main/java/by/custom/utilcalculator/service/UserProgressManager.java

```diff
@@ -1,72 +1,112 @@
 
-    public UserProgressManager() {
-        //userProgress = UserProgress.getInstance();
+    private UserProgressManager() {
         messagesCreator = MessagesCreator.getInstance();
+        fileUserProgressStorage = FileUserProgressStorage.getInstance();
     }
 
     public static UserProgressManager getInstance() {
-        return BotFieldsManagerHolder.BOT_FIELDS_MANAGER;
+        return BotFieldsManagerHolder.USER_PROGRESS_MANAGER;
     }
 
-    public String processCarOrigin(String command, UserProgress userProgress) {
-        switch (command) {
-            case Command.EAES -> userProgress.setCountryOrigin(CountryOrigin.EAES);
-            case Command.OTHER_COUNTRIES -> userProgress.setCountryOrigin(CountryOrigin.OTHER);
-        }
-        return messagesCreator.getCountryOrigin(userProgress);
+    public void createNewUserProgress(String chatID) {
+        fileUserProgressStorage.createNewUserFile(chatID);
     }
 
-    public String processOwnerType(String command, UserProgress userProgress) {
-        switch (command) {
-            case Command.JURIDICAL_PERSON -> userProgress.setOwnersType(OwnersType.JURIDICAL);
-            case Command.PHYSICAL_PERSON -> userProgress.setOwnersType(OwnersType.PHYSICAL);
+    public String processCarOrigin(String command, String chatID) {
+        UserProgress userProgress;
+        try {
+            userProgress = fileUserProgressStorage.get(chatID);
+            switch (command) {
+                case Command.EAES -> userProgress.setCountryOrigin(CountryOrigin.EAES);
+                case Command.OTHER_COUNTRIES -> userProgress.setCountryOrigin(CountryOrigin.OTHER);
+            }
+            fileUserProgressStorage.save(userProgress);
+            return messagesCreator.getCountryOrigin(userProgress);
+        } catch (UserFileNotFoundException e) {
+            return e.getMessage();
         }
-        return messagesCreator.getCountryOrigin(userProgress);
     }
 
-    public String processCarAge(String command, UserProgress userProgress) {
-        switch (command) {
-            case Command.LESS_3_YEARS_AGE -> userProgress.setCarAge(CarAge.LESS_3_YEARS);
-            case Command.BETWEEN_3_AND_7_YEARS_AGE -> userProgress.setCarAge(CarAge.BETWEEN_3_AND_7_YEARS);
-            case Command.MORE_7_YEARS_AGE -> userProgress.setCarAge(CarAge.MORE_7_YEARS);
+    public String processOwnerType(String command, String chatID) {
+        UserProgress userProgress;
+        try {
+            userProgress = fileUserProgressStorage.get(chatID);
+            switch (command) {
+                case Command.JURIDICAL_PERSON -> userProgress.setOwnersType(OwnersType.JURIDICAL);
+                case Command.PHYSICAL_PERSON -> userProgress.setOwnersType(OwnersType.PHYSICAL);
+            }
+            fileUserProgressStorage.save(userProgress);
+            return messagesCreator.getCountryOrigin(userProgress);
+        } catch (UserFileNotFoundException e) {
```

Comment:
I think it's better to move catch logic to the router. but this is a controversial decision in any case. Please, do as you feel in this case.

Reply:
It is as broad as it is long. I don't like both of this ways - they made from our pretty code some kind of spagetti-code.